File: 3514-number-of-unique-xor-triplets-ii/3514-number-of-unique-xor-triplets-ii.py
class Solution(object):
    def uniqueXorTriplets(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        
        unique_nums = list(set(nums))
        
        possible_pairs = [False] * 2048
        triplets = [False] * 2048
        
        for i in range(len(unique_nums)):
            for j in range(i, len(unique_nums)):
                possible_pairs[unique_nums[i] ^ unique_nums[j]] = True
                
        for pair_xor in range(2048):
            if possible_pairs[pair_xor]:
                for num in unique_nums:
                    triplets[pair_xor ^ num] = True
                    
        return sum(triplets)

File: 3514-number-of-unique-xor-triplets-ii/3514-number-of-unique-xor-triplets-ii.c
int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int uniqueXorTriplets(int* nums, int numsSize) {
    if (numsSize == 0) return 0;
    qsort(nums, numsSize, sizeof(int), compare);

    
    int uniqueSize = 0;
    for (int i = 0; i < numsSize; i++) {
        if (i == 0 || nums[i] != nums[i - 1]) {
            nums[uniqueSize++] = nums[i];
        }
    }

   
    bool* possible_pairs = (bool*)calloc(2048, sizeof(bool));
    bool* triplets = (bool*)calloc(2048, sizeof(bool));

    
    for (int i = 0; i < uniqueSize; i++) {
        for (int j = i; j < uniqueSize; j++) {
            int pair_xor = nums[i] ^ nums[j];
            possible_pairs[pair_xor] = true;
        }
    }

    
    for (int pair_xor = 0; pair_xor < 2048; pair_xor++) {
        if (possible_pairs[pair_xor]) {
            for (int i = 0; i < uniqueSize; i++) {
                triplets[pair_xor ^ nums[i]] = true;
            }
        }
    }

    
    int unique_triplet_count = 0;
    for (int i = 0; i < 2048; i++) {
        if (triplets[i]) {
            unique_triplet_count++;
        }
    }

    
    free(possible_pairs);
    free(triplets);

    return unique_triplet_count;
}